### DISCUSSION TITLE
Uint8Array to file I can upload in Supabase

### DISCUSSION THREAD

**Discussion by cdedreuille** [supabase/supabase#1233] [Questions] [answered]
This might be a very specific question, not related directly to Supabase but here we go. I'm working on Figma plugin and I want to save frames into images I can save in Supabse. What I get from Figma is a Uint8Array object. What would be the best way to convert it into a jpg or png image I can upload to Supabase?

**Comment by inian** [comment, marked as answer]
Hi @cdedreuille, you can use the [backend API](https://supabase.github.io/storage-api/) directly. Do Figma plugins have access to the form-data api? If not, you can polyfill that. And you can do something like this (untested)
```js
const form = new FormData();
form.append("file", new Blob([fileAsArray], {type: "image/jpg"});

  return fetch(
    `${SUPABASE_API_URL}/storage/v1/object/${SUPABASE_BUCKET_ID}/${path}`,
    {
      method: "POST",
      body: form,
      headers: {
        Authorization: `Bearer ${SUPABASE_AUTH_TOKEN}`
      },
    }
  ).then(() => {
    console.debug(`Processed file: ${path}`);
  });
```

**Reply by cdedreuille** [reply]
Thanks a lot, @inian. After a couple of tests, it works perfectly.
